unitary/alpha/quantum_world.py

```diff
@@ -106,29 +127,106 @@ def combine_with(self, other_world: "QuantumWorld"):
                 "Cannot combine sparse simulator world with non-sparse")
         self.object_name_dict.update(other_world.object_name_dict)
         self.ancilla_names.update(other_world.ancilla_names)
+        self.compiled_qubits.update(other_world.compiled_qubits)
         self.post_selection.update(other_world.post_selection)
         self.circuit = self.circuit.zip(other_world.circuit)
         # Clear effect history, since undoing would undo the combined worlds
         self.effect_history.clear()
         # Clear the other world so that objects cannot be used from that world.
         other_world.clear()
 
+    def _add_ancilla(self,
+                     namespace: str,
+                     value: Union[enum.Enum, int] = 0) -> QuantumObject:
+        """Adds an ancilla qudit object with a unique name.
+
+        Args:
+            namespace: Custom string to be added in the name
+            value: The value for the ancilla qudit
+
+        Returns:
+            The added ancilla object.
+        """
+        count = 0
+        ancilla_name = f"ancilla_{namespace}_{count}"
+        while ancilla_name in self.object_name_dict:
+            count += 1
+            ancilla_name = f"ancilla_{namespace}_{count}"
+        new_obj = QuantumObject(ancilla_name, value)
+        self.add_object(new_obj)
+        self.ancilla_names.add(ancilla_name)
+        return new_obj
+
     def _append_op(self, op: cirq.Operation):
         """Add the operation in a way designed to speed execution.
 
         For the sparse simulator post-selections should be as early as possible to cut
         down the state size. Also X's since they don't increase the size.
         """
-        if not self.use_sparse:
-            self.circuit.append(op)
-            return
 
-        if isinstance(op, PostSelectOperation) or op.gate is cirq.X:
+        if not self.use_sparse or isinstance(
+                op, PostSelectOperation) or op.gate is cirq.X:
             strategy = cirq.InsertStrategy.EARLIEST
         else:
             strategy = cirq.InsertStrategy.NEW
+
+        if self.compile_to_qubits:
+            op = self._compile_op(op)
+
         self.circuit.append(op, strategy=strategy)
 
+    def _get_num_compiled_qubits(self, qudit: cirq.Qid,
+                                 num_expected_qubits: int) -> List[cirq.Qid]:
+        """Gets the exact number of compiled qubits for the qudit.
+
+        May create extra ancilla qubits, if needed. Currently, none of the
+        non-trivial cases are exercised because our actual qubit compilation
```

Comment:
This comment is kind of confusing.  What is the "non-trivial cases" you are talking about here?  

Also, mentioning what is "exercised" is kind of weird.  I would either "exercise" it via a contrived test, or consider removing it.

Reply:
Removed the method, will work on adding it back after #77 is fixed.